src/server/lib/middleware/requestContext.ts

```diff
@@ -0,0 +1,36 @@
+import { NextFunction, Request, Response } from 'express';
+import { AsyncLocalStorage } from 'node:async_hooks';
+
+type LoggerContext = {
+	request: Request;
+	response: Response;
+};
```

Comment:
Rather than storing the entire express request and response. I think it's potentially better to limit what's available here, to avoid misuse further down the line.

If this is being used only for logging at the moment, then it should only include what's relevant for that, namely the `requestId` and `ip`, and we can always update if we need to log anything extra.

Reply:
Mm, yea thats probably a good point!